src/index.ts

```diff
@@ -3,136 +3,162 @@ import { getFileTypeFromPath, renderSvg, stringifyFunction, writeFileAsync } fro
+    const browser = await browserSource.getBrowser();
+    const page = (await browser.pages())[0];
 
-  scheduleBrowserForDestruction();
+    // ⚠️ Offline mode is enabled to prevent any HTTP requests over the network
+    await page.setOfflineMode(true);
 
-  const buffer = Buffer.from(base64, "base64");
+    // Infer the file type from the file path if no type is provided
+    if (!passedOptions.type && options.path) {
+      const fileType = getFileTypeFromPath(options.path);
 
-  if (options.path) {
-    await writeFileAsync(options.path, buffer);
-  }
+      if (config.supportedImageTypes.includes(fileType)) {
+        options.type = fileType as IOptions["type"];
+      }
+    }
 
-  if (options.encoding === "base64") {
-    return base64;
-  }
+    const base64 = await page.evaluate(stringifyFunction(renderSvg, svg, {
+      width: options.width,
+      height: options.height,
+      type: options.type,
+      quality: options.quality,
+      background: options.background,
+      clip: options.clip,
+      jpegBackground: config.jpegBackground
+    }));
 
-  if (!options.encoding) {
-    return buffer;
-  }
+    browserSource.scheduleBrowserForDestruction();
 
-  return buffer.toString(options.encoding);
-};
+    const buffer = Buffer.from(base64, "base64");
 
-const to = (svg: Buffer|string) => {
-  return async (options: IOptions): Promise<Buffer|string> => {
-    return convertSvg(svg, options);
-  };
-};
+    if (options.path) {
+      await writeFileAsync(options.path, buffer);
+    }
 
-const toPng = (svg: Buffer|string) => {
-  return async (options?: IShorthandOptions): Promise<Buffer|string> => {
-    return convertSvg(svg, {...defaultPngShorthandOptions, ...options});
-  };
-};
+    if (options.encoding === "base64") {
+      return base64;
+    }
+
+    if (!options.encoding) {
+      return buffer;
+    }
 
-const toJpeg = (svg: Buffer|string) => {
-  return async (options?: IShorthandOptions): Promise<Buffer|string> => {
-    return convertSvg(svg, {...defaultJpegShorthandOptions, ...options});
+    return buffer.toString(options.encoding);
   };
-};
+}
 
-const toWebp = (svg: Buffer|string) => {
-  return async (options?: IShorthandOptions): Promise<Buffer|string> => {
-    return convertSvg(svg, {...defaultWebpShorthandOptions, ...options});
+// tslint:disable-next-line: max-classes-per-file
+export class SvgToImg {
+  constructor (private readonly browserSource: BrowserSource) {}
+  public from (svg: Buffer|string) {
+    return new Svg(svg, this.browserSource);
   };
-};
+}
+
+const defaultBrowserSource = new BrowserSource(async () => puppeteer.launch(config.puppeteer));
 
 export const from = (svg: Buffer|string) => {
-  return {
-    to: to(svg),
-    toPng: toPng(svg),
-    toJpeg: toJpeg(svg),
-    toWebp: toWebp(svg)
-  };
-};
+  return new SvgToImg(defaultBrowserSource).from(svg);
+}
+
+/* istanbul ignore next */
+export const connect = (options?: puppeteer.ConnectOptions) => {
```

Comment:
The `connect` method should not be called without passing an `options` object so I would make that argument non optional. 

Also I would explicitly type what options should be passed to the connect method. Right now it takes everything that puppeteer does like `slowMo` and `defaultViewport`. It might be confusing to some people to see those options.

I'd explicitly type it with something like this: 

```typescript
interface ConnectOptions {
  browserWSEndpoint?: string;
  browserURL?: string; 
  ignoreHTTPSErrors?: boolean;
} 
```

Reply:
Good point. Done!